### PR TITLE
Add missing query params

### PR DESCRIPTION
**Pull request by noveogroup-amorgunov**
#### Checklist

- [ ] run `npm run test` and ~~`npm run benchmark`~~
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

#### Changes

Query params are missing when `this.prefix` don't include `:`. So query params are added only for prefix with route param like `/rest-api/:id/endpoint`.

 Fix it (move appending query params in both cases),

### CODE REVIEW

**Review by Eomm — COMMENTED** [review]
Could you add a test for this use case?

**Comment by noveogroup-amorgunov** [comment]
> Could you add a test for this use case?

Yes, sure

It's strange that the tests failed in the CI, but locally the tests were successful.

**Comment by noveogroup-amorgunov** [comment]
@Eomm hi! I write a test for this case, but I have a problem with another test "prefixed proxy with query search": 

https://github.com/fastify/fastify-http-proxy/blob/ae5cacd55a46449d8a19e27c17d8810271f7acdd/test/test.js#L871

If I add `queryParams` to dest variable (in this test `?lang=en`), this test is failed. I don't have any ideas why this happens.

We register route in this test: 

```ts
    appServer.get('/second-service', async (request, reply) => {
      return `Hello World - lang = ${request.query.lang}`
    })
```

and then send request to `/second-service?lang=en` by got:

```ts
    const resultRoot = await got(
      `${proxyAddress}/second-service?lang=en`
    )
```

Next step, plugin `handler` reply request to `dest+query`:

https://github.com/fastify/fastify-http-proxy/blob/ae5cacd55a46449d8a19e27c17d8810271f7acdd/index.js#L342

and route is not matched. If I remove query params, that all okay. Do you have any ideas why this happens?

File: test/test.js [hunk 1]
```diff
@@ -917,6 +921,28 @@ async function run () {
     const queryParams = JSON.stringify(qs.parse('foo=bar&foo=baz&abc=qux'))
     t.equal(firstProxyPrefix.body, `this is "variable-api" endpoint with id 123 and query params ${queryParams}`)
   })
+
+  test('keep the query params on proxy without params in prefix', { only: true }, async t => {
```
Comment (Eomm) [inline test/test.js:925]: If I run this test with the old code, it passes.

**Review by Eomm — CHANGES_REQUESTED** [review]
The error is:

```
{"statusCode":500,"error":"Internal Server Error","message":"source must be a relative path string"}
```

https://github.com/fastify/fastify-reply-from/blob/1ece7d7ae2568eceff008547bb625f7efc4819c6/lib/utils.js#L77


```
{
  dest: URL {
    href: 'http://[::1]:46745/second-service?lang=en',
    origin: 'http://[::1]:46745',
    protocol: 'http:',
    username: '',
    password: '',
    host: '[::1]:46745',
    hostname: '[::1]',
    port: '46745',
    pathname: '/second-service',
    search: '?lang=en',
    searchParams: URLSearchParams { 'lang' => 'en' },
    hash: ''
  }
}
```

It seems a ipv6 issue, anyway you should start from `v9.x` branch or your change will be available from fastify v5 in some months 😓

File: test/test.js (hunk 1 above)
Comment (noveogroup-amorgunov) [reply test/test.js:925]: You are right, it passes. Var `dest` from `reply.from` don't have query params in value (https://github.com/fastify/fastify-http-proxy/blob/master/index.js#L341), but they passed to upstream anyway. I suppose `reply.from` copy query params underhood

I retest my project again and it's all okay, query params passed in current version. So, I close pull request. Anyway, thank you for the review, and sorry for wasting time.

**Pull request closed without merging**